src/components/TVChartContainer/TVChartContainer.tsx

```diff
@@ -0,0 +1,172 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { TV_CHART_RELOAD_TIMESTAMP_KEY, TV_SAVE_LOAD_CHARTS_KEY } from "config/localStorage";
+import { useLocalStorage, useMedia } from "react-use";
+import { defaultChartProps, disabledFeaturesOnMobile } from "./constants";
+import useTVDatafeed from "domain/tradingview/useTVDatafeed";
+import { ChartData, IChartingLibraryWidget, IPositionLineAdapter } from "../../charting_library";
+import { getObjectKeyFromValue } from "domain/tradingview/utils";
+import { SaveLoadAdapter } from "./SaveLoadAdapter";
+import { SUPPORTED_RESOLUTIONS } from "config/tradingview";
+
+const TV_RELOAD_INTERVAL = 15 * 60 * 1000; // 15 minutes
+
+type ChartLine = {
+  price: number;
+  title: string;
+};
+
+type Props = {
+  symbol: string;
+  chainId: number;
+  savedShouldShowPositionLines: boolean;
+  chartLines: ChartLine[];
+  onSelectToken: () => void;
+  period: string;
+  setPeriod: (period: string) => void;
+};
+
+export default function TVChartContainer({
+  symbol,
+  chainId,
+  savedShouldShowPositionLines,
+  chartLines,
+  onSelectToken,
+  period,
+  setPeriod,
+}: Props) {
+  const chartContainerRef = useRef<HTMLDivElement | null>(null);
+  const tvWidgetRef = useRef<IChartingLibraryWidget | null>(null);
+  const [chartReady, setChartReady] = useState(false);
+  const [tvCharts, setTvCharts] = useLocalStorage<ChartData[] | undefined>(TV_SAVE_LOAD_CHARTS_KEY, []);
+  const { datafeed, resetCache } = useTVDatafeed();
+  const isMobile = useMedia("(max-width: 550px)");
```

Comment:
we already define `isMobile` in PositionShare component, probably better to create hook like `useIsMobile` to not repeat this specific media query

Reply:
Sure!